diff --git a/node_modules/jsforce/lib/api/metadata.js b/node_modules/jsforce/lib/api/metadata.js
index 66bf93a..1935477 100644
--- a/node_modules/jsforce/lib/api/metadata.js
+++ b/node_modules/jsforce/lib/api/metadata.js
@@ -561,6 +561,18 @@ Metadata.prototype.checkDeployStatus = function(id, includeDetails, callback) {
   }).thenCall(callback);
 };
 
+/**
+ * Cancel a deployment previously started by a call to deploy()
+ *
+ * @param {String} id - Async process id returned from deploy request
+ * @param {Callback.<Metadata~AsyncResult>} [callback] - Callback function
+ * @returns {Promise.<Metadata~AsyncResult>}
+ */
+Metadata.prototype.cancelDeploy = function(id, callback) {
+  var res = this._invoke("cancelDeploy", { asyncProcessId: id });
+  return new AsyncResultLocator(this, res).thenCall(callback);
+};
+
 
 /*--------------------------------------------*/
 
diff --git a/node_modules/jsforce/lib/connection.js b/node_modules/jsforce/lib/connection.js
index 97ed0b3..6867a7e 100644
--- a/node_modules/jsforce/lib/connection.js
+++ b/node_modules/jsforce/lib/connection.js
@@ -774,7 +774,7 @@ Connection.prototype._createParallel = function(type, records, options) {
         if (options.allOrNone || !err.errorCode) {
           throw err;
         }
-        return this._toRecordResult(null, err);
+        return self._toRecordResult(null, err);
       });
     })
   );
@@ -893,7 +893,7 @@ Connection.prototype._updateParallel = function(type, records, options) {
         if (options.allOrNone || !err.errorCode) {
           throw err;
         }
-        return this._toRecordResult(record.Id, err);
+        return self._toRecordResult(record.Id, err);
       });
     })
   );
@@ -1089,7 +1089,7 @@ Connection.prototype._destroyParallel = function(type, ids, options) {
         if (options.allOrNone || !err.errorCode) {
           throw err;
         }
-        return this._toRecordResult(id, err);
+        return self._toRecordResult(id, err);
       });
     })
   );