diff --git a/node_modules/salesforce-alm/dist/lib/auth/authCommand.js b/node_modules/salesforce-alm/dist/lib/auth/authCommand.js
index 04b8642..3755145 100644
--- a/node_modules/salesforce-alm/dist/lib/auth/authCommand.js
+++ b/node_modules/salesforce-alm/dist/lib/auth/authCommand.js
@@ -84,6 +84,7 @@ const doAuth = function (force, clientSecret, context, orgApi, loginUrl, open) {
             return !_.isNil(requestState) && requestState === query.state;
         },
         type: getType(context),
+        server: context.server,
         handleDemoModePrompt: handleDemoModePrompt.bind(context)
     };
     return (webAuthListener_1.startOauth(authListenerConfig)
