diff --git a/commands.js b/commands.js
index f31adb2142f5aeeeb6d79f3d8c8aaf872b58b2b4..b130de00c02c63faf2e20a1bce20170935a0a267 100644
--- a/commands.js
+++ b/commands.js
@@ -1,29 +1,27 @@
-module.exports = [
-  'cat',
-  'cd',
-  'chmod',
-  'cp',
-  'dirs',
-  'echo',
-  'exec',
-  'find',
-  'grep',
-  'head',
-  'ln',
-  'ls',
-  'mkdir',
-  'mv',
-  'pwd',
-  'rm',
-  'sed',
-  'set',
-  'sort',
-  'tail',
-  'tempdir',
-  'test',
-  'to',
-  'toEnd',
-  'touch',
-  'uniq',
-  'which',
-];
+require('./src/cat');
+require('./src/cd');
+require('./src/chmod');
+require('./src/cp');
+require('./src/dirs');
+require('./src/echo');
+require('./src/exec');
+require('./src/find');
+require('./src/grep');
+require('./src/head');
+require('./src/ln');
+require('./src/ls');
+require('./src/mkdir');
+require('./src/mv');
+require('./src/pwd');
+require('./src/rm');
+require('./src/sed');
+require('./src/set');
+require('./src/sort');
+require('./src/tail');
+require('./src/tempdir');
+require('./src/test');
+require('./src/to');
+require('./src/toEnd');
+require('./src/touch');
+require('./src/uniq');
+require('./src/which');
\ No newline at end of file
diff --git a/shell.js b/shell.js
index f9c6f36df9de7c10958b01d08dbaa1b20034ad8c..bfafa964e5d75509c4a2e1c0652ae5d2c6a00176 100644
--- a/shell.js
+++ b/shell.js
@@ -21,9 +21,7 @@ var common = require('./src/common');
 //@commands
 
 // Load all default commands
-require('./commands').forEach(function (command) {
-  require('./src/' + command);
-});
+require('./commands');
 
 //@
 //@ ### exit(code)