diff --git a/node_modules/tsconfig-paths-webpack-plugin/lib/plugin.js b/node_modules/tsconfig-paths-webpack-plugin/lib/plugin.js
index fb5c813..95beeb0 100644
--- a/node_modules/tsconfig-paths-webpack-plugin/lib/plugin.js
+++ b/node_modules/tsconfig-paths-webpack-plugin/lib/plugin.js
@@ -111,8 +111,9 @@ function createPluginLegacy(matchPath, resolver, absoluteBaseUrl, target, extens
     return (request, callback) => {
         const innerRequest = getInnerRequest(resolver, request);
         if (!innerRequest ||
-            innerRequest.startsWith(".") ||
-            innerRequest.startsWith("..")) {
+            !request || !request.request ||
+            request.request.startsWith(".") ||
+            request.request.startsWith("..")) {
             return callback();
         }
         matchPath(innerRequest, readJsonAsync, fileExistAsync, extensions, (err, foundMatch) => {
