diff --git a/node_modules/typescript/lib/tsc.js b/node_modules/typescript/lib/tsc.js
index 8a2ed0a..846ffe0 100644
--- a/node_modules/typescript/lib/tsc.js
+++ b/node_modules/typescript/lib/tsc.js
@@ -69892,7 +69892,7 @@ var ts;
 })(ts || (ts = {}));
 var ts;
 (function (ts) {
-    var USE_NEW_TYPE_METADATA_FORMAT = false;
+    var USE_NEW_TYPE_METADATA_FORMAT = true;
     function transformTypeScript(context) {
         var factory = context.factory, emitHelpers = context.getEmitHelperFactory, startLexicalEnvironment = context.startLexicalEnvironment, resumeLexicalEnvironment = context.resumeLexicalEnvironment, endLexicalEnvironment = context.endLexicalEnvironment, hoistVariableDeclaration = context.hoistVariableDeclaration;
         var resolver = context.getEmitResolver();
diff --git a/node_modules/typescript/lib/tsserver.js b/node_modules/typescript/lib/tsserver.js
index 814d9b3..47af475 100644
--- a/node_modules/typescript/lib/tsserver.js
+++ b/node_modules/typescript/lib/tsserver.js
@@ -83206,7 +83206,7 @@ var ts;
     /**
      * Indicates whether to emit type metadata in the new format.
      */
-    var USE_NEW_TYPE_METADATA_FORMAT = false;
+    var USE_NEW_TYPE_METADATA_FORMAT = true;
     var TypeScriptSubstitutionFlags;
     (function (TypeScriptSubstitutionFlags) {
         /** Enables substitutions for decorated classes. */
diff --git a/node_modules/typescript/lib/tsserverlibrary.js b/node_modules/typescript/lib/tsserverlibrary.js
index 64c24eb..77499c5 100644
--- a/node_modules/typescript/lib/tsserverlibrary.js
+++ b/node_modules/typescript/lib/tsserverlibrary.js
@@ -83400,7 +83400,7 @@ var ts;
     /**
      * Indicates whether to emit type metadata in the new format.
      */
-    var USE_NEW_TYPE_METADATA_FORMAT = false;
+    var USE_NEW_TYPE_METADATA_FORMAT = true;
     var TypeScriptSubstitutionFlags;
     (function (TypeScriptSubstitutionFlags) {
         /** Enables substitutions for decorated classes. */
diff --git a/node_modules/typescript/lib/typescript.js b/node_modules/typescript/lib/typescript.js
index e95b4fd..b6878f0 100644
--- a/node_modules/typescript/lib/typescript.js
+++ b/node_modules/typescript/lib/typescript.js
@@ -83400,7 +83400,7 @@ var ts;
     /**
      * Indicates whether to emit type metadata in the new format.
      */
-    var USE_NEW_TYPE_METADATA_FORMAT = false;
+    var USE_NEW_TYPE_METADATA_FORMAT = true;
     var TypeScriptSubstitutionFlags;
     (function (TypeScriptSubstitutionFlags) {
         /** Enables substitutions for decorated classes. */
diff --git a/node_modules/typescript/lib/typescriptServices.js b/node_modules/typescript/lib/typescriptServices.js
index bd9f1f6..83e7d2a 100644
--- a/node_modules/typescript/lib/typescriptServices.js
+++ b/node_modules/typescript/lib/typescriptServices.js
@@ -83400,7 +83400,7 @@ var ts;
     /**
      * Indicates whether to emit type metadata in the new format.
      */
-    var USE_NEW_TYPE_METADATA_FORMAT = false;
+    var USE_NEW_TYPE_METADATA_FORMAT = true;
     var TypeScriptSubstitutionFlags;
     (function (TypeScriptSubstitutionFlags) {
         /** Enables substitutions for decorated classes. */
diff --git a/node_modules/typescript/lib/typingsInstaller.js b/node_modules/typescript/lib/typingsInstaller.js
index b672f8c..a9e43bb 100644
--- a/node_modules/typescript/lib/typingsInstaller.js
+++ b/node_modules/typescript/lib/typingsInstaller.js
@@ -83195,7 +83195,7 @@ var ts;
     /**
      * Indicates whether to emit type metadata in the new format.
      */
-    var USE_NEW_TYPE_METADATA_FORMAT = false;
+    var USE_NEW_TYPE_METADATA_FORMAT = true;
     var TypeScriptSubstitutionFlags;
     (function (TypeScriptSubstitutionFlags) {
         /** Enables substitutions for decorated classes. */
