diff --git a/node_modules/typescript/lib/tsc.js b/node_modules/typescript/lib/tsc.js
index 78c5ec2..99655bc 100644
--- a/node_modules/typescript/lib/tsc.js
+++ b/node_modules/typescript/lib/tsc.js
@@ -71972,7 +71972,7 @@ var ts;
 })(ts || (ts = {}));
 var ts;
 (function (ts) {
-    var USE_NEW_TYPE_METADATA_FORMAT = false;
+    var USE_NEW_TYPE_METADATA_FORMAT = true;
     function transformTypeScript(context) {
         var factory = context.factory, emitHelpers = context.getEmitHelperFactory, startLexicalEnvironment = context.startLexicalEnvironment, resumeLexicalEnvironment = context.resumeLexicalEnvironment, endLexicalEnvironment = context.endLexicalEnvironment, hoistVariableDeclaration = context.hoistVariableDeclaration;
         var resolver = context.getEmitResolver();
diff --git a/node_modules/typescript/lib/tsserver.js b/node_modules/typescript/lib/tsserver.js
index f9cc0bc..389091f 100644
--- a/node_modules/typescript/lib/tsserver.js
+++ b/node_modules/typescript/lib/tsserver.js
@@ -85587,7 +85587,7 @@ var ts;
     /**
      * Indicates whether to emit type metadata in the new format.
      */
-    var USE_NEW_TYPE_METADATA_FORMAT = false;
+    var USE_NEW_TYPE_METADATA_FORMAT = true;
     var TypeScriptSubstitutionFlags;
     (function (TypeScriptSubstitutionFlags) {
         /** Enables substitutions for decorated classes. */
diff --git a/node_modules/typescript/lib/tsserverlibrary.js b/node_modules/typescript/lib/tsserverlibrary.js
index 8893292..6449e3d 100644
--- a/node_modules/typescript/lib/tsserverlibrary.js
+++ b/node_modules/typescript/lib/tsserverlibrary.js
@@ -85781,7 +85781,7 @@ var ts;
     /**
      * Indicates whether to emit type metadata in the new format.
      */
-    var USE_NEW_TYPE_METADATA_FORMAT = false;
+    var USE_NEW_TYPE_METADATA_FORMAT = true;
     var TypeScriptSubstitutionFlags;
     (function (TypeScriptSubstitutionFlags) {
         /** Enables substitutions for decorated classes. */
diff --git a/node_modules/typescript/lib/typescript.js b/node_modules/typescript/lib/typescript.js
index 85a70c4..9ebe314 100644
--- a/node_modules/typescript/lib/typescript.js
+++ b/node_modules/typescript/lib/typescript.js
@@ -85781,7 +85781,7 @@ var ts;
     /**
      * Indicates whether to emit type metadata in the new format.
      */
-    var USE_NEW_TYPE_METADATA_FORMAT = false;
+    var USE_NEW_TYPE_METADATA_FORMAT = true;
     var TypeScriptSubstitutionFlags;
     (function (TypeScriptSubstitutionFlags) {
         /** Enables substitutions for decorated classes. */
diff --git a/node_modules/typescript/lib/typescriptServices.js b/node_modules/typescript/lib/typescriptServices.js
index 668927c..3cbbc31 100644
--- a/node_modules/typescript/lib/typescriptServices.js
+++ b/node_modules/typescript/lib/typescriptServices.js
@@ -85781,7 +85781,7 @@ var ts;
     /**
      * Indicates whether to emit type metadata in the new format.
      */
-    var USE_NEW_TYPE_METADATA_FORMAT = false;
+    var USE_NEW_TYPE_METADATA_FORMAT = true;
     var TypeScriptSubstitutionFlags;
     (function (TypeScriptSubstitutionFlags) {
         /** Enables substitutions for decorated classes. */
diff --git a/node_modules/typescript/lib/typingsInstaller.js b/node_modules/typescript/lib/typingsInstaller.js
index 14d9f58..6098cf8 100644
--- a/node_modules/typescript/lib/typingsInstaller.js
+++ b/node_modules/typescript/lib/typingsInstaller.js
@@ -85576,7 +85576,7 @@ var ts;
     /**
      * Indicates whether to emit type metadata in the new format.
      */
-    var USE_NEW_TYPE_METADATA_FORMAT = false;
+    var USE_NEW_TYPE_METADATA_FORMAT = true;
     var TypeScriptSubstitutionFlags;
     (function (TypeScriptSubstitutionFlags) {
         /** Enables substitutions for decorated classes. */
