diff --git a/lib/tsc.js b/lib/tsc.js
index 5e735846308d1244cbd7722540e3c1ceab20e8d6..be9b0f89540e485ce5c0c52269911c84f283dd57 100644
--- a/lib/tsc.js
+++ b/lib/tsc.js
@@ -85950,7 +85950,7 @@ function getRawLiteral(factory2, node, currentSourceFile) {
 }
 
 // src/compiler/transformers/ts.ts
-var USE_NEW_TYPE_METADATA_FORMAT = false;
+var USE_NEW_TYPE_METADATA_FORMAT = true;
 function transformTypeScript(context) {
   const {
     factory: factory2,
diff --git a/lib/tsserver.js b/lib/tsserver.js
index 2329a12e68fc1788ad518eef084b69767f0e1a6c..1c46ce120bb1b864d0c8c0a5f1e776b6ba2c8819 100644
--- a/lib/tsserver.js
+++ b/lib/tsserver.js
@@ -90772,7 +90772,7 @@ function getRawLiteral(factory2, node, currentSourceFile) {
 }
 
 // src/compiler/transformers/ts.ts
-var USE_NEW_TYPE_METADATA_FORMAT = false;
+var USE_NEW_TYPE_METADATA_FORMAT = true;
 function transformTypeScript(context) {
   const {
     factory: factory2,
diff --git a/lib/tsserverlibrary.js b/lib/tsserverlibrary.js
index 6861715ba56445ed6982ce737fcb965176e99ba3..67e3c34315d44f70e0b70e5bd988c181d1ad8a97 100644
--- a/lib/tsserverlibrary.js
+++ b/lib/tsserverlibrary.js
@@ -90505,7 +90505,7 @@ ${lanes.join("\n")}
     "src/compiler/transformers/ts.ts"() {
       "use strict";
       init_ts2();
-      USE_NEW_TYPE_METADATA_FORMAT = false;
+      USE_NEW_TYPE_METADATA_FORMAT = true;
     }
   });
 
diff --git a/lib/typescript.js b/lib/typescript.js
index 589e2c2e3116563eb6ac510f4e3008107ffcc3ca..401645bad8bde624e6dfbbac588178fcc19a2da0 100644
--- a/lib/typescript.js
+++ b/lib/typescript.js
@@ -90505,7 +90505,7 @@ ${lanes.join("\n")}
     "src/compiler/transformers/ts.ts"() {
       "use strict";
       init_ts2();
-      USE_NEW_TYPE_METADATA_FORMAT = false;
+      USE_NEW_TYPE_METADATA_FORMAT = true;
     }
   });
 