diff --git a/node_modules/typescript/lib/tsc.js b/node_modules/typescript/lib/tsc.js
index e843aa1..8994f0b 100644
--- a/node_modules/typescript/lib/tsc.js
+++ b/node_modules/typescript/lib/tsc.js
@@ -74930,7 +74930,7 @@ var ts;
 })(ts || (ts = {}));
 var ts;
 (function (ts) {
-    var USE_NEW_TYPE_METADATA_FORMAT = false;
+    var USE_NEW_TYPE_METADATA_FORMAT = true;
     function transformTypeScript(context) {
         var factory = context.factory, emitHelpers = context.getEmitHelperFactory, startLexicalEnvironment = context.startLexicalEnvironment, resumeLexicalEnvironment = context.resumeLexicalEnvironment, endLexicalEnvironment = context.endLexicalEnvironment, hoistVariableDeclaration = context.hoistVariableDeclaration;
         var resolver = context.getEmitResolver();
diff --git a/node_modules/typescript/lib/tsserver.js b/node_modules/typescript/lib/tsserver.js
index 0283468..b00141e 100644
--- a/node_modules/typescript/lib/tsserver.js
+++ b/node_modules/typescript/lib/tsserver.js
@@ -89023,7 +89023,7 @@ var ts;
     /**
      * Indicates whether to emit type metadata in the new format.
      */
-    var USE_NEW_TYPE_METADATA_FORMAT = false;
+    var USE_NEW_TYPE_METADATA_FORMAT = true;
     var TypeScriptSubstitutionFlags;
     (function (TypeScriptSubstitutionFlags) {
         /** Enables substitutions for decorated classes. */
diff --git a/node_modules/typescript/lib/tsserverlibrary.js b/node_modules/typescript/lib/tsserverlibrary.js
index 1db4e4a..719fe9a 100644
--- a/node_modules/typescript/lib/tsserverlibrary.js
+++ b/node_modules/typescript/lib/tsserverlibrary.js
@@ -89217,7 +89217,7 @@ var ts;
     /**
      * Indicates whether to emit type metadata in the new format.
      */
-    var USE_NEW_TYPE_METADATA_FORMAT = false;
+    var USE_NEW_TYPE_METADATA_FORMAT = true;
     var TypeScriptSubstitutionFlags;
     (function (TypeScriptSubstitutionFlags) {
         /** Enables substitutions for decorated classes. */
diff --git a/node_modules/typescript/lib/typescript.js b/node_modules/typescript/lib/typescript.js
index cc9dd7d..a66f909 100644
--- a/node_modules/typescript/lib/typescript.js
+++ b/node_modules/typescript/lib/typescript.js
@@ -89217,7 +89217,7 @@ var ts;
     /**
      * Indicates whether to emit type metadata in the new format.
      */
-    var USE_NEW_TYPE_METADATA_FORMAT = false;
+    var USE_NEW_TYPE_METADATA_FORMAT = true;
     var TypeScriptSubstitutionFlags;
     (function (TypeScriptSubstitutionFlags) {
         /** Enables substitutions for decorated classes. */
diff --git a/node_modules/typescript/lib/typescriptServices.js b/node_modules/typescript/lib/typescriptServices.js
index 1f85972..16d8e19 100644
--- a/node_modules/typescript/lib/typescriptServices.js
+++ b/node_modules/typescript/lib/typescriptServices.js
@@ -89217,7 +89217,7 @@ var ts;
     /**
      * Indicates whether to emit type metadata in the new format.
      */
-    var USE_NEW_TYPE_METADATA_FORMAT = false;
+    var USE_NEW_TYPE_METADATA_FORMAT = true;
     var TypeScriptSubstitutionFlags;
     (function (TypeScriptSubstitutionFlags) {
         /** Enables substitutions for decorated classes. */
diff --git a/node_modules/typescript/lib/typingsInstaller.js b/node_modules/typescript/lib/typingsInstaller.js
index 17e9323..11fdc99 100644
--- a/node_modules/typescript/lib/typingsInstaller.js
+++ b/node_modules/typescript/lib/typingsInstaller.js
@@ -89012,7 +89012,7 @@ var ts;
     /**
      * Indicates whether to emit type metadata in the new format.
      */
-    var USE_NEW_TYPE_METADATA_FORMAT = false;
+    var USE_NEW_TYPE_METADATA_FORMAT = true;
     var TypeScriptSubstitutionFlags;
     (function (TypeScriptSubstitutionFlags) {
         /** Enables substitutions for decorated classes. */
