patch-package
--- a/node_modules/vlocity/lib/datapacksutils.js
+++ b/node_modules/vlocity/lib/datapacksutils.js
@@ -630,7 +630,9 @@ DataPacksUtils.prototype.runApex = function(projectPath, filePaths, currentConte
                         callback();
                     }
                 });
-        });
+            }).catch(err => {
+                callback(err);
+            });
     }, function(err, result) {
         if (err) {
             reject(err);
