diff --git a/node_modules/vlocity/lib/datapacks.js b/node_modules/vlocity/lib/datapacks.js
index 6aabe7a..41a974f 100644
--- a/node_modules/vlocity/lib/datapacks.js
+++ b/node_modules/vlocity/lib/datapacks.js
@@ -217,6 +217,11 @@ DataPacks.prototype.getStorageLimitExceededError = function() {
 DataPacks.prototype.runDataPackProcess = async function(dataPackData, options) {
     var self = this;
 
+    if (options.cancellationToken && options.cancellationToken.isCancellationRequested) {
+        VlocityUtils.warn('Detected cancellation requested. Skipping further job processing.');
+        throw new Error('Operation Cancelled');
+    }
+
     if (options && dataPackData && dataPackData.processData) {
 
         Object.keys(options).forEach(function(optionKey){
@@ -279,6 +284,11 @@ DataPacks.prototype.runDataPackProcess = async function(dataPackData, options) {
 
             let result = await self.makeApexPostRequest(self.dataPacksEndpoint(), dataPackData);
 
+            if (options.cancellationToken && options.cancellationToken.isCancellationRequested) {
+                VlocityUtils.warn('Detected cancellation requested. Skipping further job processing.');
+                return { VlocityDataPackId: dataPackId, message: 'Operation Cancelled', dataPackError: true, Status: 'Cancelled' };
+            }
+
             VlocityUtils.verbose('Result', result);
 
             if (!result) {
@@ -316,23 +326,24 @@ DataPacks.prototype.runDataPackProcess = async function(dataPackData, options) {
                 result.Total = 1;
             }
 
-            if (soloExportDataPackKey 
-                && (!options.maxDepth 
-                    || options.maxDepth == -1)
-                && (result.Finished > 0 
-                    && result.Status != "Ready" 
-                    && result.Status != "Error" 
-                    && !result.Async)) {
-                result.Status = "Complete";
-            } else if (!result.Async 
-                && dataPackData.processType == "Export" 
-                && result.Status == "InProgress"
-                && (!options.maxDepth 
-                    || options.maxDepth == -1)
-                && options.exportPacksMaxSize 
-                && (result.Total - startingExportSize) > options.exportPacksMaxSize) {
-                    result.Status = "Complete";
-            }
+            // Disable this as it causes dependencies to be ommited resulting in partial exports.
+            // if (soloExportDataPackKey 
+            //     && (!options.maxDepth 
+            //         || options.maxDepth == -1)
+            //     && (result.Finished > 0 
+            //         && result.Status != "Ready" 
+            //         && result.Status != "Error" 
+            //         && !result.Async)) {
+            //     result.Status = "Complete";
+            // } else if (!result.Async 
+            //     && dataPackData.processType == "Export" 
+            //     && result.Status == "InProgress"
+            //     && (!options.maxDepth 
+            //         || options.maxDepth == -1)
+            //     && options.exportPacksMaxSize 
+            //     && (result.Total - startingExportSize) > options.exportPacksMaxSize) {
+            //         result.Status = "Complete";
+            // }
 
             if (result.Total > 0 && result.Async && result.Total == result.Finished) {
                 result.Finished--;
diff --git a/node_modules/vlocity/lib/datapacksbuilder.js b/node_modules/vlocity/lib/datapacksbuilder.js
index 535f9d2..76359be 100644
--- a/node_modules/vlocity/lib/datapacksbuilder.js
+++ b/node_modules/vlocity/lib/datapacksbuilder.js
@@ -1,7 +1,6 @@
-var yaml = require('js-yaml');
 var path = require('path');
 var fs = require('fs-extra');
-var sass = require('sass.js');
+var sass = null;
 var stringify = require('fast-json-stable-stringify');
 
 var UTF8_EXTENSIONS = [ "css", "json", "yaml", "scss", "html", "js", "xml"];
@@ -14,7 +13,7 @@ var DataPacksBuilder = module.exports = function(vlocity) {
 
     this.compileQueue = []; // array with files that require compilation
 
-    this.defaultdatapack = fs.readFileSync(path.join(__dirname, 'defaultdatapack.json'), 'utf8');
+    this.defaultdatapack = JSON.stringify(require('./defaultdatapack.json'));
 
     this.dataPackSizes = {};
 
@@ -232,10 +231,7 @@ DataPacksBuilder.prototype.buildImport = async function(importPath, jobInfo, val
 
         if (jobInfo.upgradeDataPackFields) {
             jobInfo.platformObjectMappings = this.vlocity.datapacksutils.updateExpandedDefinitionNamespace(
-                yaml.safeLoad(fs.readFileSync(
-                    path.join(__dirname, 'platformObjectMappings.yaml'),
-                    'utf8'
-                ))
+                require('./platformObjectMappings.yaml')
             );
             dataPackImport.dataPacks.forEach(dataPack => {
                 if (Object.keys(jobInfo.platformObjectMappings).includes(dataPack.VlocityDataPackType)) {
@@ -416,33 +412,52 @@ DataPacksBuilder.prototype.getFileData = function() {
 }
 
 DataPacksBuilder.prototype.setFileData = async function(filePath, encoding) {
-    var data = await fs.readFile(filePath, encoding);
+    //const data = await fs.readFile(filePath, encoding);
+    const fileKey = path.normalize(filePath).toLowerCase();
+    const isDatapackMeta = fileKey.endsWith('_datapack.json') || fileKey.endsWith('.json');
 
     if (!this.allFileDataMap) {
         this.allFileDataMap = {};
     }
 
-    this.allFileDataMap[path.normalize(filePath).toLowerCase()] = data;
-
-    try {
-        var dataPack = JSON.parse(data);
-
-        if (dataPack) {
-            if (!this.recordSourceKeyToFilePath) {
-                this.recordSourceKeyToFilePath = {};
+    // Define transparent lazy property on the all file datamap
+    Object.defineProperty(this.allFileDataMap, fileKey, {
+        get() {
+            // Initlize lazy loader cache on allFileDataMap obj
+            const cache = this['__cache'] || (this['__cache'] = new Map());
+            let fileData = cache.get(fileKey);
+            if (!fileData) {
+                // Ideally we make this an async function but for now
+                // keep it sync for simplicty and backawrd compaitiblity
+                fileData = fs.readFileSync(filePath, encoding);
+                cache.set(fileKey, fileData);
             }
+            return fileData;
+        }
+    });
             
-            if (dataPack instanceof Array) {
-                for (data of dataPack) {
-                    if (data.VlocityRecordSourceKey) {
-                        this.recordSourceKeyToFilePath[data.VlocityRecordSourceKey] = path.normalize(filePath);    
+    if (isDatapackMeta) {
+        try {
+            var dataPack = JSON.parse(this.allFileDataMap[fileKey]);
+            if (dataPack) {
+                if (!this.recordSourceKeyToFilePath) {
+                    this.recordSourceKeyToFilePath = {};
+                }
+                
+                if (dataPack instanceof Array) {
+                    for (data of dataPack) {
+                        if (data.VlocityRecordSourceKey) {
+                            this.recordSourceKeyToFilePath[data.VlocityRecordSourceKey] = path.normalize(filePath);    
+                        }   
                     }   
+                } else if (dataPack.VlocityRecordSourceKey){
+                    this.recordSourceKeyToFilePath[dataPack.VlocityRecordSourceKey] = path.normalize(filePath);
                 }
-            } else if (dataPack.VlocityRecordSourceKey){
-                this.recordSourceKeyToFilePath[dataPack.VlocityRecordSourceKey] = path.normalize(filePath);
             }
+        } catch (e) {
+            VlocityUtils.error(`Unable to parse ${path.basename(filePath)} >> ${e.message || e}`);
         }
-    } catch (e) {}
+    }
 }
 
 DataPacksBuilder.prototype.loadFilesAtPath = async function(srcpath, jobInfo, dataPackKey) {
@@ -1297,6 +1312,11 @@ DataPacksBuilder.prototype.compile = function(lang, source, options, cb) {
     // add addtional languages here to support more compilation types    
     switch(lang) {
         case 'scss': {
+            // Lazy load SASS module; sass module is rather heavy; lazy loading it has a benefit
+            // as to not load the full 3MB JS code up front but instead only when required for the first time 
+            // by our datapack builder
+            sass = (sass || require('sass.js'));
+
             // intercept file loading requests from libsass
             sass.importer((request, done) => {
                 // (object) request
diff --git a/node_modules/vlocity/lib/datapackserrorhandling.js b/node_modules/vlocity/lib/datapackserrorhandling.js
index 26cb86f..cae7464 100644
--- a/node_modules/vlocity/lib/datapackserrorhandling.js
+++ b/node_modules/vlocity/lib/datapackserrorhandling.js
@@ -1,10 +1,6 @@
-var fs = require("fs-extra");
-var path = require('path');
-var yaml = require('js-yaml');
-
 var DataPacksErrorHandling = module.exports = function(vlocity) {
     this.vlocity = vlocity || {};
-    this.errorHandlingDefinition = yaml.safeLoad(fs.readFileSync(path.join(__dirname, "datapackserrorhandling.yaml"), 'utf8'));
+    this.errorHandlingDefinition =  require("./datapackserrorhandling.yaml");
 };
 
 DataPacksErrorHandling.prototype.getSanitizedErrorMessage = async function(jobInfo, dataPack) {
diff --git a/node_modules/vlocity/lib/datapacksjob.js b/node_modules/vlocity/lib/datapacksjob.js
index 7879216..b45c50f 100644
--- a/node_modules/vlocity/lib/datapacksjob.js
+++ b/node_modules/vlocity/lib/datapacksjob.js
@@ -9,8 +9,8 @@ const open = require("opn");
 var DataPacksJob = module.exports = function(vlocity) {
     this.vlocity = vlocity || {};
 
-    this.defaultJobSettings = yaml.safeLoad(fs.readFileSync(path.join(__dirname, 'defaultjobsettings.yaml'), 'utf8'));
-    this.queryDefinitions = yaml.safeLoad(fs.readFileSync(path.join(__dirname, 'querydefinition.yaml'), 'utf8'));
+    this.defaultJobSettings = require('./defaultjobsettings.yaml');
+    this.queryDefinitions = require('./querydefinition.yaml');
     this.runningParallel = {};
     this.vlocity.relationMap = new Map();
     this.vlocity.insertIndexToSfIdMap = new Map();
@@ -28,7 +28,8 @@ var SUPPORTED_JOB_KEY_TO_OPTION_MAP = {
     source: 'source',
     alreadyExportedKeys: 'alreadyExportedKeys',
     exportPacksMaxSize: 'exportPacksMaxSize',
-    useVlocityTriggers: 'useVlocityTriggers'
+    useVlocityTriggers: 'useVlocityTriggers',
+    cancellationToken: 'cancellationToken'
 };
 
 var MAX_PER_GROUP = 10;
@@ -1315,8 +1316,10 @@ DataPacksJob.prototype.exportGroup = async function(inputMap) {
             return false;
         }
 
-        VlocityUtils.success('Exporting', dataPack.VlocityDataPackType, this.vlocity.datapacksutils.getDisplayName(dataPack));
-
+        var name = this.vlocity.datapacksutils.getDisplayName(dataPack);
+        if (name) {
+            VlocityUtils.success('Exporting', dataPack.VlocityDataPackType, name);
+        }
         return true;
     });
 
@@ -2478,7 +2481,8 @@ DataPacksJob.prototype.deployPack = async function(inputMap) {
             throw e;
         }
 
-        VlocityUtils.error('Error', 'Deploying Pack', e.stack);
+        VlocityUtils.error('Error', 'Deploying Pack', e.stack || e);
+        throw e;
     }
 }
 
diff --git a/node_modules/vlocity/lib/datapacksutils.js b/node_modules/vlocity/lib/datapacksutils.js
index a96735c..f4fc60a 100644
--- a/node_modules/vlocity/lib/datapacksutils.js
+++ b/node_modules/vlocity/lib/datapacksutils.js
@@ -26,7 +26,7 @@ var DataPacksUtils = module.exports = function(vlocity) {
 
     CURRENT_INFO_FILE = path.join(vlocity.tempFolder, 'currentJobInfo.json');
 
-    this.dataPacksExpandedDefinition = yaml.safeLoad(fs.readFileSync(path.join(__dirname, 'datapacksexpanddefinition.yaml'), 'utf8'));
+    this.dataPacksExpandedDefinition = require('./datapacksexpanddefinition.yaml');
 
     this.runJavaScriptModules = {};
 
@@ -825,6 +825,7 @@ DataPacksUtils.prototype.loadApex = async function(projectPath, filePath) {
     var possiblePaths = [
         path.join(projectPath, filePath),
         path.join(__dirname, '..', 'apex', filePath),
+        path.join(__dirname, 'apex', filePath),
         filePath
     ];
 
@@ -1240,6 +1241,9 @@ DataPacksUtils.prototype.getDisplayName = function(dataPack) {
             dataPackKey = dataPack.VlocityDataPackKey.substring(dataPack.VlocityDataPackKey.indexOf('/') + 1);
         } else if (dataPack.VlocityDataPackKeyForManifest) {
             dataPackKey = dataPack.VlocityDataPackKeyForManifest.substring(dataPack.VlocityDataPackKeyForManifest.indexOf('/') + 1);
+            if (dataPackKey == 'null') {
+                dataPackKey = null;
+            }
         } else if (dataPack.VlocityRecordSourceKey) {
             dataPackKey = dataPack.VlocityRecordSourceKey.substring(dataPack.VlocityRecordSourceKey.indexOf('/') + 1);
         }
@@ -2537,7 +2541,7 @@ DataPacksUtils.prototype.printJobStatus = function(jobInfo, forceWrite) {
 DataPacksUtils.prototype.updateStatusIPC = function(logInfo) {
     let ipcRenderer;
     if ('electron' in process.versions) {
-        ipcRenderer = require('electron').ipcRenderer;
+        //ipcRenderer = require('electron').ipcRenderer;
     } else {
         return;
     }
diff --git a/node_modules/vlocity/lib/utilityservice.js b/node_modules/vlocity/lib/utilityservice.js
index 718875b..de6b5b9 100644
--- a/node_modules/vlocity/lib/utilityservice.js
+++ b/node_modules/vlocity/lib/utilityservice.js
@@ -1,5 +1,4 @@
 var jsforce = require('jsforce');
-var yaml = require('js-yaml');
 var fs = require('fs-extra');
 var path = require('path');
 var stringify = require('fast-json-stable-stringify');
@@ -482,7 +481,7 @@ UtilityService.prototype.getPackageVersion = async function() {
     if (!this.vlocity.PackageVersion) {
         var result = await this.vlocity.jsForceConnection.query("SELECT DurableId, Id, IsSalesforce, MajorVersion, MinorVersion, Name, NamespacePrefix FROM Publisher where NamespacePrefix = \'" + this.vlocity.namespace + "\' LIMIT 1");
 
-        this.vlocity.buildToolsVersionSettings = yaml.safeLoad(fs.readFileSync(path.join(__dirname, "buildToolsVersionSettings.yaml"), 'utf8'));
+        this.vlocity.buildToolsVersionSettings = require('./buildToolsVersionSettings.yaml');
 
         this.vlocity.BuildToolSettingLatestVersion = this.vlocity.buildToolsVersionSettings.latest;
 
diff --git a/node_modules/vlocity/lib/validationtest.js b/node_modules/vlocity/lib/validationtest.js
index d85f750..6d7ed7e 100644
--- a/node_modules/vlocity/lib/validationtest.js
+++ b/node_modules/vlocity/lib/validationtest.js
@@ -17,7 +17,7 @@ ValidationTest.prototype.validate = async function(jobInfo, currentContextData)
     };
 
     self.jobInfo = jobInfo;
-    self.config = yaml.safeLoad(fs.readFileSync(path.join(__dirname, 'validationtestsettings.yaml'), 'utf8'));
+    self.config = require('./validationtestsettings.yaml');
     self.vlocityMatchingKeys = await self.vlocity.utilityservice.getDRMatchingKeyFields();
     var allTests = self.loadAllTests();
     var runTestData = await self.groupRunTestData(currentContextData, allTests);
diff --git a/node_modules/vlocity/lib/vlocityutils.js b/node_modules/vlocity/lib/vlocityutils.js
index 9b76a81..2434806 100644
--- a/node_modules/vlocity/lib/vlocityutils.js
+++ b/node_modules/vlocity/lib/vlocityutils.js
@@ -60,7 +60,7 @@ VlocityUtils.getIPCRenderer = function() {
 
     if (ipcRendererSingleton === undefined) {
         if ('electron' in process.versions) {
-            ipcRendererSingleton = require('electron').ipcRenderer;
+            ipcRendererSingleton = false;
         } else {
             ipcRendererSingleton = false;
         }
