patch-package
--- a/node_modules/vlocity/lib/utilityservice.js
+++ b/node_modules/vlocity/lib/utilityservice.js
@@ -375,7 +375,7 @@ UtilityService.prototype.getPackageVersion = async function() {
     
     VlocityUtils.verbose('Get Package Version');
 
-    if (!this.vlocity.packageVersion) {
+    if (!this.vlocity.PackageVersion) {
         var result = await this.vlocity.jsForceConnection.query("SELECT DurableId, Id, IsSalesforce, MajorVersion, MinorVersion, Name, NamespacePrefix FROM Publisher where NamespacePrefix = \'" + this.vlocity.namespace + "\' LIMIT 1");
 
         this.vlocity.buildToolsVersionSettings = yaml.safeLoad(fs.readFileSync(path.join(__dirname, "buildToolsVersionSettings.yaml"), 'utf8'));
